File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/presenter/FavoritesPresenter.java
package com.github.batulovandrey.unofficialurbandictionary.presenter;

import com.github.batulovandrey.unofficialurbandictionary.adapter.DefinitionAdapter;
import com.github.batulovandrey.unofficialurbandictionary.bean.DefinitionResponse;

import java.util.List;

/**
 * @author Andrey Batulov on 30/10/2017
 */

public interface FavoritesPresenter {

    List<DefinitionResponse> getFavorites();

    void clearList();

    void showAlertDialog();

    void showToast(int resId);

    DefinitionAdapter createNewDefinitionAdapter(List<DefinitionResponse> favorites);

    DefinitionAdapter getDefinitionAdapter();

    void hideRecycler();

    void showRecycler();
}

File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/presenter/FavoritesPresenter.kt
package com.github.batulovandrey.unofficialurbandictionary.presenter

import com.github.batulovandrey.unofficialurbandictionary.adapter.DefinitionAdapter
import com.github.batulovandrey.unofficialurbandictionary.bean.DefinitionResponse

/**
 * @author Andrey Batulov on 22/12/2017
 */

interface FavoritesPresenter {

    val favorites: List<DefinitionResponse>

    val definitionAdapter: DefinitionAdapter

    fun clearList()

    fun showAlertDialog()

    fun showToast(resId: Int)

    fun createNewDefinitionAdapter(favorites: List<DefinitionResponse>): DefinitionAdapter

    fun hideRecycler()

    fun showRecycler()
}